### ISSUE TITLE
[Npcap] Multiple UAC requests when starting/using Wireshark

### ISSUE THREAD

**Issue by aircrack-ng** [nmap/npcap#99]
I updated to the latest available release (Npcap 0.07 r17) and checked the option to only allow admin user to use it. When starting Wireshark, I had about 10 requests one after the other from UAC for NPcapHelper. Every time capture is started, it also pops up.

It would be great if there was no more than a single request.

**Comment by guyharris** [comment]
> Same issue here, every time I start Wireshark I have 7-8 UAC Windows (because of several adapters, some of which seem useless -- "NdisWan Adapter").

Useless adapters is a way of life with libpcap:

    $ tcpdump -D
    1.en0 [Up, Running]
    2.p2p0 [Up, Running]
    3.awdl0 [Up, Running]
    4.llw0 [Up, Running]
    5.utun0 [Up, Running]
    6.utun1 [Up, Running]
    7.vmnet1 [Up, Running]
    8.vmnet8 [Up, Running]
    9.lo0 [Up, Running, Loopback]
    10.bridge0 [Up, Running]
    11.en4 [Up, Running]
    12.en6 [Up, Running]
    13.en7 [Up, Running]
    14.en8 [Up, Running]
    15.gif0 [none]
    16.stf0 [none]
    17.ap1 [none]

(macOS).

> Extremely annoying, especially that UAC (unlike sudo) has no timer. Sometimes I want to quickly capture some traffic (because something is happening) and I have to wait lots of time to pass all these UAC Windows one by one, even with keyboard shortcuts it takes time as Windows takes time to display them... Every elevation request is a prompt. I think the solution is to start a single elevated process that can then spawn many as required. But indeed, this change should probably be done in Wireshark rather than npcap.

Yes, Aa least some of this is a Wireshark issue, not an Npcap issue, as per [Wireshark bug 15082](https://bugs.wireshark.org/bugzilla/show_bug.cgi?id=15082).  I presume that a single process that opens multiple adapters won't get a UAC prompt for every adapter; if so, then, if dumpcap were to have a mode where it opened multiple adapters when running in "sparkline mode" - the mode where it's capturing on an interface and supplying packet counts so that the Wireshark main window can draw sparklines for each adapter - rather than, as per that bug, having *multiple* dumpcaps, one per interface - that would at least reduce the number of UAC prompts.

**Comment by Prasannakumar-kumta** [comment]
I just uninstalled Ncap and installed winPcap. Did I do any mistake?

**Comment by JonLevin25** [comment]
This would be great - better to have it admin-only, but man are the string of UAC prompts annoying.

**Comment by clearlinkit** [comment]
TBH I'm not entirely sure what the use for the admin-only restriction is for.  So, I'm not positive if this resolves the issue or not.  But I found if I just launch Wireshark with run as administrator the UAC prompt does not occur.  Other than the original one to authorize launching as administrator.

**Comment by clearlinkit** [comment]
TBH I'm not entirely sure what the use for the admin-only restriction is for. So, I'm not positive if this resolves the issue or not. But I found if I just launch Wireshark with run as administrator the UAC prompt does not occur. Other than the original one to authorize launching as administrator.

**Comment by johnthacker** [comment]
I did update Wireshark (but this won't come out until 4.4.0) to reduce the number of UAC pop-ups by getting all the interface capability information at once, along with a few other changes to reduce the pop-ups. There can still be pop-ups for:
1. Retrieving the entire list.
2. Retrieving the capabilities of selected interfaces in the list (this could be folded into the first request, but there's a little bit to workaround for possibly querying twice interfaces that can be put into monitor-mode, which may support different link-layer types with and without monitor mode enabled).
3. And of course actually capturing.

There's also a preference (which can be set at the command like with `-o capture.no_interface_load:TRUE`) that avoids loading the entire list of interfaces at startup.

**Comment by guyharris** [comment]
> 3. And of course actually capturing.

And to make things clearer, that means:

3a. The sparklines on the main screen and the Capture Options dialog.
3b. Actually capturing.

**Comment by johnthacker** [comment]
> And to make things clearer, that means:
> 
> 3a. The sparklines on the main screen and the Capture Options dialog. 3b. Actually capturing.

Yes. The sparklines don't appear if the list isn't loaded, so right now it's possible to get the number of UAC prompts to two, with

`Wireshark -k -o capture_no_interface.load:TRUE` and either having previously specified a default device, or passing `-i N` where `N` is the number of an interface, or `-i <name>` where `<name>` can be the friendly name or UUID based name of a device. That's once for the capabilities of the interface, and once to start capturing.
